### ISSUE TITLE
Injecting from `<img>` should takes its attributes

### ISSUE THREAD

**Issue by sntran** [tanem/svg-injector#32]
For example, `width`, `height`, `alt`, `title`, etc...

This is probably opinionated. For me, if I use the SVG in `<img>`, and I set `width` and `height`, I would think that the injected SVG should keep the same dimension. Otherwise, when using in IE10-, if the SVG itself does not have `width` and `height`, IE will display it using the default dimension.

**Comment by sntran** [comment]
Also, I would love to have `src` attribute remain on the injected SVG, because I need to calculate path on `use` elements inside it.

**Comment by tanem** [comment]
> For example, width, height, alt, title, etc...
>
> This is probably opinionated. For me, if I use the SVG in <img>, and I set width and height, I would think that the injected SVG should keep the same dimension. Otherwise, when using in IE10-, if the SVG itself does not have width and height, IE will display it using the default dimension.

`id`, `title` & `style` should already be coming through from the `img`. I can look at adding `width` and `height` support though.

In terms of setting `width` and `height` now, will it work for your use-case in the meantime to make use of that `style` attribute transfer? Something like `<img style="width: 10px; height: 10px" ...`.

> Also, I would love to have src attribute remain on the injected SVG, because I need to calculate path on use elements inside it.

Putting `src` on SVG would be invalid markup IIRC. Would be keen to see some more detail here though as I don't quite understand what the use-case is. Maybe a working example or a code snippet?

**Comment by sntran** [comment]
> In terms of setting width and height now, will it work for your use-case in the meantime to make use of that style attribute transfer? Something like <img style="width: 10px; height: 10px" ....

Don't worry about that, there are different workarounds :)

> Putting src on SVG would be invalid markup IIRC.

Agree. Probably `data-src` or some other attribute.

> Would be keen to see some more detail here though as I don't quite understand what the use-case is.

My use case is the `<use>` element inside the SVG. Let's say I have the following SVG:`<img src="/content/inline-images/icon-1.svg" />`

```svg
<svg viewBox="0 0 100 100">
   <use href="defs.svg#icon-1"></use>
</svg>
```

The `defs.svg` contains list of symbols and is in the same directory relative to the current SVG file, i.e., `/content/inline-images/defs.svg`. After injecting the SVG, the `href="defs.svg"` is not valid and won't be loaded. In my `each` callback, I need to replace the `href` with the actual path relative to the site's base URL, relative to the `icon-1.svg`. Therefore, I will need to know the `src` to do the calculation.

Don't feel obligated to add such support though. I can always workaround for my special use cases. I thought it would be nicer to have the library support it :)

**Comment by tanem** [comment]
Thanks for the detailed response, it really helps 🙏

There's a way we can get that information across without making any changes actually. Since [all `data-` attributes are transferred from the `img`](https://github.com/tanem/svg-injector/blob/master/src/inject-element.ts#L91), would adding a `data-src` attribute to your `img` tags solve your issue?

Something like:

```html
<img src="/content/inline-images/icon-1.svg" data-src="/content/inline-images/icon-1.svg" />
```

You can call the `data-` attribute whatever you want since they're all taken across anyway. So if it's easier for your calculation, you could use something like `data-baseurl="/content/inline-images/"`.

**Comment by sntran** [comment]
That's essentially what I did. My images come from a CMS though, so they don't have `data-src`. I just had to use JS to add the attribute before injecting.

**Comment by tanem** [comment]
Ah, I see how it would be useful in the library then. I don't think it'll be a big change, so I'll have a crack at it today sometime :+1:

**Comment by tanem** [comment]
Just opened #33 ... let me know if you have any feedback, I'll merge it later today most likely.

**Comment by tanem** [comment]
Released in `5.0.6` :+1:

**Comment by tanem** [comment]
I'll close this down now as I think we have resolved the relevant issues from above? Feel free to tag me in a comment though if you think that's not the case, and we can re-open if need be.

**Comment by sntran** [comment]
I guess you will look at `width` and `height` attributes at another time then. That's fine with me. Thanks so much for working on this.

**Comment by tanem** [comment]
@sntran oh... I thought when you said this:

> Don't worry about that, there are different workarounds :)

... that you were talking about `width` and `height` 😆 

I'll take a look at moving those attrs over now anyway since I'm online ⏳

**Comment by sntran** [comment]
I was. But they're "workarounds" :) it's better if the library support it.

I'm a little reluctant though. I know I want those attributes for my use case, but not sure if they are desired in every cases.

**Comment by tanem** [comment]
Released as `6.0.0`. Came to a similar conclusion re whether this is actually useful for most cases, check the PR notes for details.

Let me know when you're happy to close the issue down anyway, I'll leave it open for now.

**Comment by sntran** [comment]
Agreed on your notes. Having an option for this would probably be more flexible, but I'm okay to close this issue.
